cylc/sphinx_ext/cylc_lang/autodocumenters.py

```diff
@@ -346,3 +367,314 @@ def run(self):
+            capture_output=True,
+        )
+        CylcMetadataDirective.check_subproc_output(sub)
+        return json.loads(sub.stdout)
+
+    @staticmethod
+    def get_workflow_metadata(config, conf_path):
+        """Get workflow metadata.
+
+        Additional processing:
+        * If no title field is provided use either the workflow folder
+          or the task/family name.
+        * Don't return the root family if there is no metadata.
+
+        Returns:
+            'workflow': {.. top level metadata ..},
+            'runtime': {'namespace': '.. task or family metadata ..'}
+
+        """
+        # Extract Data
+        meta = {}
+
+        # Copy metadata to the two top level sections:
+        meta['workflow'] = config.get('meta')
+        meta['runtime'] = {
+            k: v.get('meta', {})
+            for k, v in config.get('runtime', {}).items()}
+
+        # Title is parent directory if otherwise unset:
+        if not meta.get('workflow', {}).get('title', ''):
+            meta['workflow']['title'] = Path(conf_path).name
+
+        # Title of namespace is parent if otherwise unset:
+        poproot = False
+        for namespace, info in meta['runtime'].items():
+            # don't display root unless it's actually had some
+            # metadata added, but save a flag rather than modifying
+            # the iterable being looped over:
+            if (
+                namespace == 'root'
+                and not any(meta['runtime'].get('root').values())
+            ):
+                poproot = True
+
+            # If metadata doesn't have a title set title to the namespace name:
+            if not info.get('title', ''):
+                meta['runtime'][namespace]['title'] = namespace
+
+        if poproot:
+            meta['runtime'].pop('root')
+
+        return meta
+
+    @staticmethod
+    def workflow_to_node(meta, src):
+        """Convert workflow metadata to list of strings for use as a node.
+
+        """
+        ret = []
+
+        # Handle the workflow config metadata:
+        workflow_title = meta.get('workflow', {}).get('title', src)
+        workflow_desc = meta.get('workflow', {}).get(
+            'description', '').split('\n')
+
+        ret += [f'.. cylc:conf:: {workflow_title}', '']
+        ret += [f'   {t.strip()}' for t in workflow_desc] + ['']
+
+        # Handle the runtime config metadata:
+        ret += [f'{INDENT}Runtime', f'{INDENT}=======', '']
+
+        for taskmeta in meta['runtime'].values():
+            indent = INDENT
+            title = taskmeta["title"]
+            ret += [f'{indent}.. cylc:conf:: {title}', '']
+            indent += INDENT
+
+            url = taskmeta.get('url', '')
+            if url:
+                ret += [
+                    f'{indent}.. seealso::',
+                    '',
+                    f'{indent + INDENT}{url}',
+                    ''
+                ]
+
+            desc = taskmeta.get('description', '').replace('\n', ' ')
```

Comment:
Why flatten this? This content will be interpreted as ReST so only a double line break would cause the output content to break onto a new line.

User's might want to use ReST in their description.